### ISSUE TITLE
[ADD] Rendering using Osmarender (XSLT -> SVG)

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#694]
**[Submitted to the original trac issue database at 6.28pm, Sunday, 17th February 2008]**

The idea would be to use libxslt to render the data with the Osmarender .xslt
libxslt is available as win32 binaries at (http://www.zlatkovic.com/libxml.en.html).
Don't know if available for MAC

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 11.31am, Sunday, 16th March 2008]**

I have a +/- working POC. Unfortunately, QT support of SVG is a bit buggy.
Out-of-the-box, a SVG created from osmarender shows practically nothing, including on QT 4.4 beta.

A bug has been submitted and accepted (as task 202426) by trolltech. This, at least, shows roads.


```
--- src/svg/qsvghandler.cpp.old	2008-03-07 12:19:24.057690900 +0100
+++ src/svg/qsvghandler.cpp	2008-03-07 11:19:46.595352000 +0100
@@ -954,7 +954,7 @@
     //qDebug()<<"Node "<<node->type()<<", attrs are "<<value<<width;

 

     if (!value.isEmpty() || !width.isEmpty() || !linecap.isEmpty() ||

-        linejoin.isEmpty()) {

+        !linejoin.isEmpty()) {

         if (value != QLatin1String("none")) {

             QSvgStrokeStyle *inherited =

                 static_cast<QSvgStrokeStyle*>(node->styleProperty(

```

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 1.57pm, Thursday, 7th August 2008]**

(In [9541]) ADD : Render to SVG via Osmarender (requires libxml2/libxslt; inkscape required to generate bitmap). Closes #694
ADD : Render to bitmap via native engine.
ADD : layer menu in menubar
ADD : display viewport coordinates in status bar
ADD : (almost) generalize progress dialogs
ADD : support (initial) of GPS devices / NMEA log replay
ADD : option to show/hide track segments
ADD : option to show/hide non-stylized trackpoints
ADD : option to show/hide street names on the map
